Commonality Analysis Code.Rmd

```diff
@@ -0,0 +1,258 @@
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_language = fct_lump_min(b_dem_language, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_language_10_text = fct_lump_min(b_dem_language_10_text, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_zip_code = fct_lump_min(b_dem_zip_code, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_grade = fct_lump_min(b_dem_grade, 10)) 
+cope_ca_data <- cope_ca_data %>%
+  mutate(b_dem_grade_9_text = fct_lump_min(b_dem_grade_9_text, 10)) 
+str(cope_ca_data)
+```
+
+```{r data imputation}
+set.seed(12345) #use the same seed number each time to make the code reproducible (the seed number is just the starting point used in the generation of a sequence of random numbers in the random forest)
+
+cope_ca_data_imputed <- missForest(cope_ca_data)
+#NRMSE = 0.25. This is the error term for the continuous variables.
+#PFC = 0.04. This is the error term for the categorical variables.
+#We will run the analyses again with complete cases only (no imputation) as a robustness check to make sure there isn't a huge difference in results.
+```
+
+```{r merge imputed data}
+#Turn the imputed data into it's own object
+cope_ca_data_imputed <- as.data.frame(cope_ca_data_imputed$ximp)
+
+#Select out the imputed data of the variables that we actually want imputed values for, from cope_ca_data_imputed: "pi_bhs", "pi_shs", "f1_bhs", "f1_shs", "f1_cdi", "f1_gad", "f1_cts_rs", "f1_drs_month"
+cope_ca_data_imputed <- cope_ca_data_imputed %>%
+  select(contains(c("pi_bhs", "pi_shs", "f1_bhs", "f1_shs", "f1_cdi", "f1_gad", "f1_cts_rs", "f1_drs_month")))
+  
+#Select out the raw data of the variables that we do not want imputed values for, from cope_ca_data (assuming that lumping factors doesn't cause issues): "b_dem","b_bhs", "b_shs", "b_cdi", "b_gad", "b_cts_rs","b_drs_month", "pi_pfs", "condition", "pi_dropout"
+cope_ca_data_dem_b_pfs_condition_completion <- cope_ca_data %>%
+  select(contains(c("b_dem","b_bhs", "b_shs", "b_cdi", "b_gad", "b_cts_rs","b_drs_month", "pi_pfs", "condition", "pi_dropout")))
+
+#Merge the imputed data with the demographics, baseline, program feedback, condition, and completion variables 
+cope_ca_data_imputed <- cbind(cope_ca_data_imputed, cope_ca_data_dem_b_pfs_condition_completion)
+#1383 observations which is a bit less than the 1385 we expected in the pre-registration
+
+#Since PFS was not imputed but we need PFS completers for the analyses, look for people who skipped one or more PFS item
+missing_pfs <- cope_ca_data_imputed [, c("pi_pfs_1", "pi_pfs_2", "pi_pfs_3", "pi_pfs_4", "pi_pfs_5", "pi_pfs_6", "pi_pfs_7")]
+vis_miss(missing_pfs, show_perc_col = T, sort_miss = T)
+miss_var_summary(missing_pfs)
+#Looks like 1 participant (row 782) did not answer item 2, so need to drop them
+cope_ca_data_imputed <- cope_ca_data_imputed[-782, ]
+#Now at #1382 observations
+```
+
+```{r create new data frame of complete cases (nonimputed) - for robustness check later}

```

Comment:
Not really sure what you mean because we are not using it earlier? Maybe it would have been more relevant if we kept the part where I was merging datasets

Reply:
I think you could use it when you're selecting columns from `cope_full_data`, right? NBD either way